getReducedProblem.m

```diff
@@ -0,0 +1,144 @@
+function [newParameters,newProperties,newObjectiveFunction,newOptions] = getReducedProblem(parameters,properties,objectiveFunction,options)
+
+    % Definition of index set of optimized parameters
+    freePars = setdiff(1:parameters.number, options.fixedParameters);
+    fixedPars = options.fixedParameters;
+    fixedParValues = options.fixedParameterValues;
+   
+    %% Reduced parameter struct
+    % Basic information
+    newParameters.name = parameters.name(freePars);
+    newParameters.min = parameters.min(freePars);
+    newParameters.max = parameters.max(freePars);
+    newParameters.guess = parameters.guess(freePars);
+    newParameters.number = length(freePars);
+    if isfield(parameters, 'constraints')
+        if isfield(parameters.constraints, 'A')
+            if ~isempty(parameters.constraints.A)
+                newParameters.constraints.A = parameters.constraints.A(:,freePars);
+            else
+                newParameters.constraints.A = [];
+            end
+        else
+            newParameters.constraints.A = [];
+        end
+
+        if isfield(parameters.constraints, 'b')
+            if ~isempty(parameters.constraints.b)
+                newParameters.constraints.b = parameters.constraints.b - parameters.constraints.A(:,options.fixedParameters)*options.fixedParameterValues;
+            else
+                newParameters.constraints.b = [];
+            end
+        else
+            newParameters.constraints.A = [];
+        end
+
+        if isfield(parameters.constraints, 'Aeq')
+            if ~isempty(parameters.constraints.Aeq)
+                newParameters.constraints.Aeq = parameters.constraints.Aeq(:,freePars);
+            else
+                newParameters.constraints.Aeq = [];
+            end
+        else
+            newParameters.constraints.Aeq = [];
+        end
+
+        if isfield(parameters.constraints, 'beq')
+            if ~isempty(parameters.constraints.beq)
+                newParameters.constraints.beq = parameters.constraints.beq - parameters.constraints.Aeq(:,options.fixedParameters)*options.fixedParameterValues;
+            else
+                newParameters.constraints.beq = [];
+            end
+        else
+            newParameters.constraints.beq = [];
+        end
+    end
+
+    % Multi-start optimization results
+    if isfield(parameters,'MS')
```

Comment:
I would argue that if we do this for `MS`, we should also to this for `S` and `P`.